src/LBPairUpgradeableBeacon.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: GPL-3
+
+pragma solidity 0.8.20;
+
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+import {BaseUpgradeableBeacon} from "./BaseUpgradeableBeacon.sol";
+import {ILBFactory} from "./interfaces/ILBFactory.sol";
+import {IAccessControl} from "@openzeppelin/contracts/access/IAccessControl.sol";
+
+contract LBPairUpgradeableBeacon is BaseUpgradeableBeacon {
+    error Beacon__UnauthorizedCaller(address caller);
+    error Beacon__IsPaused();
+    error Beacon__IsNotPaused();
+    error Beacon__NoPausedImplementation();
+
+    address private _pausedImplementation;
+    ILBFactory private _lbFactoryAddress;
+
+    // @notice this is a determenistic PausedTarget contract address that is used as the beacon implementation when it's paused. The contract will revert on any function call. It is used because we cant utilize non-contract (i.e. 0x0 or 0x1) as the implementation of the beacon.  
+    address public constant TARGET_PAUSED_CONTRACT_ADDRESS = 0xE919920aE49d3027566025548f12cDaB4E52b595;
+
+    modifier onlyAuthorized() {
+        if(msg.sender != owner() && !IAccessControl(address(_lbFactoryAddress)).hasRole(_lbFactoryAddress.getPauserRole(), msg.sender)) revert Beacon__UnauthorizedCaller(msg.sender);
+        _;
+    }
+
+    modifier whenNotPaused() {
+        if(isPaused()) revert Beacon__IsPaused();
+        _;
+    }
+
+    modifier whenPaused() {
+        if(!isPaused()) revert Beacon__IsNotPaused();
+        _;
+    }
+
+    constructor (address _implementation, address _owner, address _factoryAddress) BaseUpgradeableBeacon(_implementation, _owner) {
+        if(_owner != address(0)) {
+            _transferOwnership(_owner);
+        }
```

Comment:
this is already called in the BaseUpgradeableBeacon constructor

Reply:
Updated in this [commit](https://github.com/DistributedCollective/sovryn-lb-dex/pull/3/commits/e7e76f053b2778191a3461452b40d2f0dfe647e7)